### PR TITLE
Remove taskref/pipelineref deprecated bundle field

### LINKED ISSUE #7411: Deprecate the legacy OCI bundles implementation
The OCI bundles feature is implemented today through remote resolution, which is in beta.
The feature flag for OCI bundles is still in alpha. It should be deprecated and eventually removed. 

The OCI bundles features will mature as part of the overall remote resolution work.

Related work: https://github.com/tektoncd/pipeline/issues/5514

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This field has been deprecated for about a year and half. So this is
removing this field from v1beta1 (it's not present in v1 already).

Closes #7411

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
`taskRef.bundle` and `pipelineRef.bundle` are now removed from `v1beta1` API version, as they were deprecated for about a year and half. Using them will result in a error when creating an object.

action required: make sure you migrate from `taskRef.bundle` and `pipelineRef.bundle` to the bundle resolver (see https://tekton.dev/docs/pipelines/migrating-v1beta1-to-v1/#replacing-taskrefbundle-and-pipelinerefbundle-with-bundle-resolver-a-idreplacing-taskrefbundle-and-pipelinerefbundle-with-bundle-resolver-a)
```

### CODE REVIEW

**Comment by vdemeester** [comment]
@afrittoli @chitrangpatel @JeromeJu

File: docs/pipelines.md [hunk 1]
```diff
@@ -585,45 +585,24 @@ There is currently a hard limit of 20 objects in a bundle.
...
-
-{{% tab "v1beta1" %}}
 ```yaml
 spec:
   tasks:
     - name: hello-world
-      taskRef:
-        name: echo-task
-        bundle: docker.com/myrepo/mycatalog
+	  taskRef:
```
Comment (JeromeJu) [inline docs/pipelines.md:592]: NIT: tabs look a bit off here?

**Review by JeromeJu — COMMENTED** [review]
Thanks @vdemeester , this looks generally good and thanks for helping with the codebase hygiene - just one question do we want to tombstone the field? Having this on my mind because perviously there were several fields regarding `PipelineResources` brought back for tombstone after getting removed. 🤔

**Comment by vdemeester** [comment]
> Thanks @vdemeester , this looks generally good and thanks for helping with the codebase hygiene - just one question do we want to tombstone the field? Having this on my mind because perviously there were several fields regarding `PipelineResources` brought back for tombstone after getting removed. 🤔

Ah good point, that might be a problem for results or chains, so we could "keep" it in code, but do nothing.
cc @tektoncd/results-maintainers @tektoncd/chains-maintainers

**Comment by lcarva** [comment]
There's some context here: https://github.com/tektoncd/pipeline/issues/6430


> > Thanks @vdemeester , this looks generally good and thanks for helping with the codebase hygiene - just one question do we want to tombstone the field? Having this on my mind because perviously there were several fields regarding `PipelineResources` brought back for tombstone after getting removed. 🤔
> 
> Ah good point, that might be a problem for results or chains, so we could "keep" it in code, but do nothing. cc @tektoncd/results-maintainers @tektoncd/chains-maintainers

+1 removing the field can be problematic which is what we saw with PipelineResources. @wlynch explains the intricacies here: https://github.com/tektoncd/pipeline/issues/6430

I believe it comes down to, let's prevent resources from being created with this attribute on release X. Then, when release X is the oldest supported release, we can safely drop the attribute.

**Comment by vdemeester** [comment]
> I believe it comes down to, let's prevent resources from being created with this attribute on release X. Then, when release X is the oldest supported release, we can safely drop the attribute.

Yes, that definitely make sense. We could start disallowing the field in the next release (or the release that ships this PR) while keeping the field for "clients" like chains…

**Comment by vdemeester** [comment]
Updated with : 
- Keeping the field in the go code
- Disallowing it at creation time (so that updates of existing object with the bundle wouldn't fail), does that make sense ?

**Comment by JeromeJu** [comment]
Looks like we'd also need to get rid of the feature flag

**Comment by vdemeester** [comment]
/retest

@lcarva @wlynch @chitrangpatel @JeromeJu 🙏🏼

**Review by JeromeJu — APPROVED** [review]

**Review by chitrangpatel — APPROVED** [review]
/lgtm

**Pull request merged**